recipes/qpdf/all/patches/0001-libqpdf-cmake-deps-jpeg-zlib.patch

```diff
@@ -0,0 +1,25 @@
+diff --git a/libqpdf/CMakeLists.txt b/libqpdf/CMakeLists.txt
+index 7053e205..8fd72c11 100644
+--- a/libqpdf/CMakeLists.txt
++++ b/libqpdf/CMakeLists.txt
+@@ -403,6 +403,11 @@ endif()
+ # use PIC for the object library so we don't have to compile twice.
+ set(OBJECT_LIB libqpdf_object)
+ add_library(${OBJECT_LIB} OBJECT ${libqpdf_SOURCES})
++find_package(JPEG REQUIRED)
++find_package(ZLIB REQUIRED)
++target_link_libraries(${OBJECT_LIB} PUBLIC JPEG::JPEG)
++target_link_libraries(${OBJECT_LIB} PUBLIC ZLIB::ZLIB)
++
+ if(OBJECT_LIB_IS_PIC)
+   target_compile_definitions(${OBJECT_LIB} PRIVATE libqpdf_EXPORTS)
+ endif()
+@@ -498,8 +503,6 @@ if(BUILD_SHARED_LIBS)
+     PUBLIC
+     $<BUILD_INTERFACE:${qpdf_INCLUDE}>
+     $<INSTALL_INTERFACE:include>)
+-  target_link_directories(${SHARED_LIB} PRIVATE ${dep_link_directories})
+-  target_link_libraries(${SHARED_LIB} PRIVATE ${dep_link_libraries})
```

Comment:
You will also need to be sure to use `target_link_libraries(${SHARED_LIB} PUBLIC JPEG::JPEG ZLIB::ZLIB)` here and likewise for `STATIC_LIB` further down. Note that the shared / static libraries may not be built from the corresponding `OBJECT_LIB`, in which case, the dependencies won't be propagated to the target as they should be.

Reply:
The code here is the issue:

```
  if(OBJECT_LIB_IS_PIC)
    add_library(${SHARED_LIB} SHARED $<TARGET_OBJECTS:libqpdf_object>)
  else()
    add_library(${SHARED_LIB} SHARED ${libqpdf_SOURCES})
  endif()
```

`OBJECT_LIB_IS_PIC` will be false because `BUILD_STATIC_LIBS` is set to false when building shared libraries. Therefore, the shared library is not built from the object library and does not receive the link libraries from the `target_link_libraries` calls on the `OBJECT_LIB`.

Reply:
Yeah you are right. I hoped the object lib will transport it automatically as a public dependency. Nevertheless, to make sure it is really consumed by the other targets, i will now just hard-code add it to them also.